lib/Service/CardService.php

```diff
@@ -390,6 +391,29 @@ public function update($id, $title, $stackId, $type, $owner, $description = '',
 		return $card;
 	}
 
+	public function cloneCard(int $id, ?int $targetStackId = null):Card {
+		$this->permissionService->checkPermission($this->cardMapper, $id, Acl::PERMISSION_READ);
+		$originCard = $this->cardMapper->find($id);
+		if ($targetStackId == null) {
+			$targetStackId = $originCard->getStackId();
+		}
+		$this->permissionService->checkPermission($this->stackMapper, $targetStackId, Acl::PERMISSION_EDIT);
+		$newCard = $this->create($originCard->getTitle(), $targetStackId, $originCard->getType(), $originCard->getOrder(), $originCard->getOwner());
+		$boardId = $this->stackMapper->findBoardId($targetStackId);
+		foreach ($this->labelMapper->findAssignedLabelsForCard($id) as $label) {
+			if ($boardId != $this->stackMapper->findBoardId($originCard->getStackId())) {
+				$label = $this->labelService->cloneLabelIfNotExists($label->getId(), $boardId);
+			}
+			$this->assignLabel($newCard->getId(), $label->getId());
+		}
+		foreach ($this->assignedUsersMapper->findAll($id) as $assignement) {
+			$this->assignmentService->assignUser($newCard->getId(), $assignement->getParticipant());
```

Comment:
I'm unsure if we should only assign users that have access to the target board. Might be confusing otherwise if you see people assigned but they don't have access to the card.